pkg/services/secrets/kvstore/remote_plugin.go

```diff
@@ -163,3 +168,45 @@ func updateFatalFlag(ctx context.Context, skv secretsKVStorePlugin) {
 func wrapUserFriendlySecretError(ufe string) datasources.ErrDatasourceSecretsPluginUserFriendly {
 	return datasources.ErrDatasourceSecretsPluginUserFriendly{Err: ufe}
 }
+
+func GetNamespacedKVStore(kv kvstore.KVStore) *kvstore.NamespacedKVStore {
+	return kvstore.WithNamespace(kv, kvstore.AllOrganizations, PluginNamespace)
+}
+
+func isPluginStartupErrorFatal(ctx context.Context, kvstore *kvstore.NamespacedKVStore) (bool, error) {
+	_, exists, err := kvstore.Get(ctx, QuitOnPluginStartupFailureKey)
+	if err != nil {
+		return false, errors.New(fmt.Sprint("error retrieving key ", QuitOnPluginStartupFailureKey, " from kvstore. error: ", err.Error()))
+	}
+	return exists, nil
+}
+
+func setPluginStartupErrorFatal(ctx context.Context, kvstore *kvstore.NamespacedKVStore, isFatal bool) error {
+	if !isFatal {
+		return kvstore.Del(ctx, QuitOnPluginStartupFailureKey)
+	}
+	return kvstore.Set(ctx, QuitOnPluginStartupFailureKey, "true")
+}
+
+func shouldUseRemoteSecretsPlugin(mg plugins.SecretsPluginManager, cfg *setting.Cfg) bool {
```

Comment:
See comment on kvstore.go. We could potentially just do away with this all together by just always starting the plugin if installed, and then just using the use_plugin flag for everything else.

Reply:
Wouldn't it save resources to not start the plugin if it's not going to be used?

Reply:
The case I could see would be if we need to trigger a migration from the plugin later on, like if we add an API call to do it. In the case, we could just make sure the plugin is started at that time.